atc/db/resource.go

```diff
@@ -953,3 +956,243 @@ func requestScheduleForJobsUsingResource(tx Tx, resourceID int) error {
 
 	return nil
 }
+
+var (
+	downStreamCausalityQuery = `
+WITH RECURSIVE build_ids AS (
+		SELECT DISTINCT i.build_id
+			FROM build_resource_config_version_inputs i
+			WHERE i.resource_id=$1 AND i.version_md5=$2
+	UNION ALL
```

Comment:
nit: maybe we can do just a UNION instead of a UNION ALL? Probably because we don't need duplicate resulting build_ids right